packages/access-mocsy/package.py

```diff
@@ -6,24 +6,54 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack.package import find_libraries, install, join_path, mkdirp
+from spack.build_systems import cmake, makefile
 
-# https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html
-class AccessMocsy(MakefilePackage):
+
+class AccessMocsy(CMakePackage, MakefilePackage):
     """Routines to model ocean carbonate system thermodynamics. ACCESS NRI's fork."""
 
     homepage = "https://www.access-nri.org.au"
     git = "https://github.com/ACCESS-NRI/mocsy.git"
 
-    maintainers("harshula")
+    maintainers("harshula", "dougiesquire")
+
+    # https://github.com/ACCESS-NRI/mocsy/blob/master/LICENSE
+    license("MIT", checked_by="dougiesquire")
+
+    version("gtracers", branch="gtracers")
 
-    version("master", branch="master")
-    version("gtracers", branch="gtracers", preferred=True)
-    # TODO: Remove the 'mom5' version. The 'gtracers' version/branch is a copy of 'mom5'.
-    version("mom5", branch="mom5")
+    build_system("makefile", "cmake", default="cmake")
+
+    with when("build_system=cmake"):
+        variant(
+            "build_type",
+            default="RelWithDebInfo",
+            description="CMake build type",
+            values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel"),
+        )
+        variant(
+            "precision",
+            default="2",
+            description="Precision to use (1 or 2)",
+            values=("1", "2"),
+        )
 
-    # Need: self.spec["mpi"].mpifc
     depends_on("mpi")
 
+    flag_handler = build_system_flags
```

Comment:
After what @edoyango found, should we avoid overriding the default flag handler?

Reply:
Let's leave it here for now and we can bulk update all relevant SPRs if that's the way we decide to go